tests/app.rs

```diff
@@ -1,39 +1,30 @@
-use infer::{Infer, MatcherType, Type};
+use infer::{MatcherType, Type};
 
-fn matcher(_buf: &[u8]) -> bool {
-    false
-}
+mod common;
 
-#[test]
-fn test_elf() {
-    let info = Infer::new();
+test_format!(
+    MatcherType::APP,
+    "application/x-executable",
+    "elf",
+    test_elf,
```

Comment:
It's a bit ugly the fact that I have to pass the full function name but I don't think there's a stable way of doing something like concatenating `test_` with `elf`.

Reply:
Indeed, [what you are looking for](https://github.com/rust-lang/rust/issues/29599) is AFAIU unstable for now.

I understand your frustration, but as long as this is not in the public api this can be easily changed once we have something better (I have not so far).